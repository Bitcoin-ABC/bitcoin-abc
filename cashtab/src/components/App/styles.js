// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled, { createGlobalStyle, css } from 'styled-components';
import { ToastContainer } from 'react-toastify';

export const ExtensionFrame = createGlobalStyle`
    html, body {
        min-width: 400px;
        min-height: 600px;
    }
`;

export const GlobalStyle = createGlobalStyle`
    *::placeholder {
        color: ${props => props.theme.forms.placeholder} !important;
    }
    a {
        color: ${props => props.theme.eCashBlue}
        &:hover {
            color: ${props => props.theme.eCashPurple}
            text-decoration: none;
        }
    }
`;

export const CashtabNotification = styled(ToastContainer)`
    .Toastify__progress-bar-theme--dark {
        background: #00abe7;
    }
    .Toastify__progress-bar-theme--light {
        background: #00abe7;
    }
`;

export const CustomApp = styled.div`
    text-align: center;
    font-family: 'Poppins', sans-serif;
    background-color: ${props => props.theme.backgroundColor};
    background-size: 100px 171px;
    background-image: ${props => props.theme.backgroundImage};
    background-attachment: fixed;
    min-height: 100vh;
    box-sizing: border-box;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;
export const WalletBody = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
`;
const AppPositionCss = css`
    width: 500px;
    background: ${props => props.theme.walletBackground};
    -webkit-box-shadow: 0px 0px 24px 1px ${props => props.theme.shadow};
    -moz-box-shadow: 0px 0px 24px 1px ${props => props.theme.shadow};
    box-shadow: 0px 0px 24px 1px ${props => props.theme.shadow};
    @media (max-width: 768px) {
        width: 100%;
        -webkit-box-shadow: none;
        -moz-box-shadow: none;
        box-shadow: none;
    }
`;

export const WalletCtn = styled.div`
    ${AppPositionCss}
    position: relative;
    padding: 0 0 100px;
    min-height: calc(100vh - 80px);
`;

export const Footer = styled.div`
    ${AppPositionCss}
    z-index: 2;
    height: 80px;
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    position: fixed;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0;
`;

export const NavWrapper = styled.div`
    width: 100%;
    height: 100%;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 1.3rem;
    margin-bottom: 5px;
`;

export const NavIcon = styled.span`
    @media (hover: hover) {
        ${NavWrapper}:hover & {
            background-color: ${props =>
                props.clicked ? 'transparent' : props.theme.eCashPurple};
            ::before,
            ::after {
                background-color: ${props => props.theme.eCashPurple};
            }
        }
    }

    position: relative;
    background-color: ${props =>
        props.clicked ? 'transparent' : props.theme.buttons.primary.color};
    width: 2rem;
    height: 2px;
    display: inline-block;
    transition: transform 300ms, top 300ms, background-color 300ms;
    &::before,
    &::after {
        content: '';
        background-color: ${props => props.theme.buttons.primary.color};
        width: 2rem;
        height: 2px;
        display: inline-block;
        position: absolute;
        left: 0;
        transition: transform 300ms, top 300ms, background-color 300ms;
    }
    &::before {
        top: ${props => (props.clicked ? '0' : '-0.8rem')};
        transform: ${props => (props.clicked ? 'rotate(135deg)' : 'rotate(0)')};
    }
    &::after {
        top: ${props => (props.clicked ? '0' : '0.8rem')};
        transform: ${props =>
            props.clicked ? 'rotate(-135deg)' : 'rotate(0)'};
    }
`;

export const NavMenu = styled.div`
    position: fixed;
    margin-right: 125px;
    bottom: 80px;
    display: flex;
    width: 225px;
    flex-direction: column;
    border: ${props => (props.open ? '1px solid' : '0px solid')};
    border-color: ${props =>
        props.open ? props.theme.contrast : 'transparent'};
    justify-content: center;
    align-items: center;
    @media (max-width: 768px) {
        right: 0;
        margin-right: 0;
    }
    overflow: hidden;
    transition: ${props =>
        props.open
            ? 'max-height 250ms ease-in-out , border-color 250ms ease-in-out, border-width 250ms ease-in-out'
            : 'max-height 250ms cubic-bezier(0, 1, 0, 1), border-color 250ms ease-in-out, border-width 250ms ease-in-out'};
    max-height: ${props => (props.open ? '100vh' : '0')};
`;

export const NavItem = styled.button`
    display: flex;
    justify-content: space-between;
    text-align: left;
    font-size: 24px;
    padding: 12px;
    align-items: center;
    width: 100%;
    white-space: nowrap;
    background-color: ${props => props.theme.walletBackground};
    border: 1px solid ${props => props.theme.walletBackground};
    color: ${props => props.theme.contrast};
    gap: 6px;
    cursor: pointer;
    &:hover {
        color: ${props => props.theme.eCashPurple};
        svg,
        g,
        path {
            fill: ${props => props.theme.eCashPurple};
        }
    }
    svg {
        fill: ${props => props.theme.contrast};
        max-width: 33px;
        height: auto;
        flex: 1;
    }
    g,
    path {
        fill: ${props => props.theme.contrast};
    }
    p {
        flex: 2;
        margin: 0;
    }
    ${({ active, ...props }) =>
        active &&
        `    
        color: ${props.theme.navActive};
        svg, g, path {
            fill: ${props.theme.navActive};
        }
  `}
`;

export const NavButton = styled.button`
    :focus,
    :active {
        outline: none;
    }
    @media (hover: hover) {
        :hover {
            svg,
            g,
            path {
                fill: ${props => props.theme.eCashPurple};
            }
        }
    }
    width: 100%;
    height: 100%;
    cursor: pointer;
    padding: 0;
    background: none;
    border: none;
    font-size: 10px;
    svg {
        fill: ${props => props.theme.contrast};
        width: 30px;
        height: 30px;
    }
    g,
    path {
        fill: ${props => props.theme.contrast};
    }
    ${({ active, ...props }) =>
        active &&
        `    
        color: ${props.theme.navActive};
        svg, g, path {
            fill: ${props.theme.navActive};
        }
  `}
`;
export const ScreenWrapper = styled.div`
    padding: 0px 30px;
    @media (max-width: 768px) {
        padding: 0px 15px;
    }
`;

export const HeaderCtn = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding: 15px 0;
    gap: 6px;
`;

export const CashtabLogo = styled.img`
    width: 120px;
    @media (max-width: 768px) {
        width: 110px;
    }
`;

// Easter egg styled component not used in extension/src/components/App.js
export const EasterEgg = styled.img`
    position: fixed;
    bottom: -195px;
    margin: 0;
    right: 10%;
    transition-property: bottom;
    transition-duration: 1.5s;
    transition-timing-function: ease-out;

    :hover {
        bottom: 0;
    }

    @media screen and (max-width: 1250px) {
        display: none;
    }
`;

export const NavHeader = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 1rem;
    color: ${props => props.theme.navActive};
    svg {
        padding: 0.2rem;
        fill: ${props => props.theme.navActive};
        height: 33px;
        width: 30px;
    }
    g {
        fill: ${props => props.theme.navActive};
    }
    path {
        fill: ${props => props.theme.navActive};
    }
`;

export const BalanceHeaderContainer = styled.div`
    ${props =>
        props.minified
            ? `background-image: linear-gradient(to bottom, ${props.theme.walletBackground}, ${props.theme.walletInfoContainer})`
            : `background: ${props.theme.walletInfoContainer}`};
    padding: 12px 20px;
    padding-top: 0px;
    box-sizing: border-box;
    transition: all 0.5s ease-in-out;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;
