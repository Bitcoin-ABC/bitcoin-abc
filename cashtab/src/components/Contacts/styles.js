// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const ContactList = styled.div`
    margin-top: 24px;
    padding: 12px;
    display: flex;
    flex-direction: column;
    width: 100%;
    align-items: center;
    gap: 12px;
    color: ${props => props.theme.contrast};
    box-sizing: border-box;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;

export const ContactsPanel = styled.div`
    display: flex;
    flex-direction: column;
    padding: 12px;
    width: 100%;
    background-color: ${props => props.theme.panel};
    border-radius: 9px;
    margin-bottom: 12px;
`;

export const Row = styled.div`
    display: flex;
    width: 100%;
    align-items: center;
    justify-content: space-between;
    gap: 12px;
    flex-wrap: wrap;
    border-bottom: 0.5px solid ${props => props.theme.separator};
    padding: 6px 0;
`;
export const ButtonRow = styled.div`
    display: flex;
    width: 100%;
    align-items: center;
    gap: 12px;
`;

export const ContactListName = styled.div`
    display: flex;
    text-align: left;
    word-break: break-word;
`;

export const ButtonPanel = styled.div`
    display: flex;
    gap: 9px;
    align-items: baseline;
`;
