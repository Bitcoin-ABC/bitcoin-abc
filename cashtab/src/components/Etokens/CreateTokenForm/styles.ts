// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const Form = styled.div`
    display: flex;
    flex-direction: column;
    gap: 12px;
`;
export const SwitchRow = styled.div`
    display: flex;
    flex-direction: row;
    gap: 12px;
    align-items: center;
`;
export const SwitchLabel = styled.div`
    text-align: left;
    color: ${props => props.theme.contrast};
    font-size: 18px;
`;
export const EditIcon = styled.div`
    cursor: pointer;
    color: ${props => props.theme.contrast};
    &:hover {
        color: ${props => props.theme.eCashBlue};
    }
    word-wrap: break-word;
`;
export const TokenCreatedLink = styled.a`
    color: ${props => props.theme.walletBackground};
    text-decoration: none;
`;
export const IconModalForm = styled.div`
    display: flex;
    flex-direction: column;
    gap: 12px;
    justify-content: center;
`;
export const IconModalRow = styled.div`
    display: flex;
    width: 100%;
    gap: 3px;
`;
export const SliderLabel = styled.div`
    color: ${props => props.theme.contrast};
`;
export const SliderBox = styled.div`
    width: 100%;
`;
export const CropperContainer = styled.div`
    height: 200px;
    position: relative;
`;
export const CreateTokenTitle = styled.h3`
    color: ${props => props.theme.contrast};
`;
export const TokenCreationSummaryTable = styled.div`
    color: ${props => props.theme.contrast};
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    gap: 3px;
`;
export const SummaryRow = styled.div`
    display: flex;
    gap: 12px;
    justify-content: flex-start;
    align-items: start;
    width: 100%;
`;
export const TokenParam = styled.div`
    word-break: break-word;
`;
export const ButtonDisabledMsg = styled.div`
    font-size: 14px;
    color: ${props => props.theme.forms.error};
    word-break: break-all;
`;
