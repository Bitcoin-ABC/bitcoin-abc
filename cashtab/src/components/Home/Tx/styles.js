// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';

export const TxWrapper = styled.div`
    border-bottom: 0.5px solid ${props => props.theme.separator};
    display: flex;
    flex-direction: column;
    gap: 12px;
    svg {
        width: 33px;
        height: 33px;
    }
    img {
        height: 33px;
    }
    box-sizing: border-box;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;
export const Collapse = styled.div`
    display: flex;
    flex-direction: column;
    gap: 12px;
    cursor: pointer;
`;
const Incoming = css`
    color: ${props => props.theme.eCashBlue};
    fill: ${props => props.theme.eCashBlue};
`;
const Genesis = css`
    color: ${props => props.theme.genesisGreen};
    fill: ${props => props.theme.genesisGreen};
    svg {
        fill: ${props => props.theme.genesisGreen};
    }
    path {
        fill: ${props => props.theme.genesisGreen};
    }
    g {
        fill: ${props => props.theme.genesisGreen};
    }
`;
const Burn = css`
    color: ${props => props.theme.eCashPurple};
    fill: ${props => props.theme.eCashPurple};
    svg {
        fill: ${props => props.theme.eCashPurple};
    }
    path {
        fill: ${props => props.theme.eCashPurple};
    }
    g {
        fill: ${props => props.theme.eCashPurple};
    }
`;
export const MainRow = styled.div`
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    align-items: center;
    gap: 12px;
    width: 100%;
    color: ${props => props.theme.contrast};
    fill: ${props => props.theme.contrast};
    ${props =>
        (props.type === 'Received' ||
            props.type === 'Staking Reward' ||
            props.type === 'Coinbase Reward') &&
        Incoming}
`;
export const MainRowLeft = styled.div`
    display: flex;
    align-items: center;
    gap: 12px;
`;
export const AppTxIcon = styled.div``;
export const TxDescCol = styled.div`
    flex-direction: row;
`;
// Top row of TxDescCol
export const TxDescSendRcvMsg = styled.div`
    display: inline-block;
`;
export const TxDesc = styled.div`
    display: flex;
    flex-wrap: wrap;
    text-align: left;
    width: 100%;
    align-items: center;
    gap: 6px;
`;
// Bottom row of TxDescCol
export const Timestamp = styled.div`
    display: flex;
    width: 100%;
    text-align: left;
    color: ${props => props.theme.lightWhite};
`;
export const AmountCol = styled.div`
    flex-direction: row;
    justify-content: flex-end;
`;
// Top row of TxAmountCol
export const AmountTop = styled.div`
    display: flex;
    width: 100%;
    justify-content: flex-end;
`;
export const AmountBottom = styled.div`
    display: flex;
    width: 100%;
    color: ${props => props.theme.lightWhite};
    justify-content: flex-end;
`;
export const CashtabMsg = styled.div`
    display: flex;
    width: 100%;
`;
export const TokenEntry = styled.div`
    display: flex;
    width: 100%;
`;
// Button panel for actions on each tx
export const Expand = styled.div`
    display: flex;
    overflow: hidden;
    height: ${props => (props.showPanel ? '36px' : '0px')};
    visibility: ${props => (props.showPanel ? 'visible' : 'collapse')};
    transition: all 0.5s ease-out;
    justify-content: space-between;
    align-items: center;
    gap: 12px;
    svg {
        height: 33px;
        width: 33px;
        fill: ${props => props.theme.contrast};
    }
    path {
        fill: ${props => props.theme.contrast};
    }
    g {
        fill: ${props => props.theme.contrast};
    }
`;
export const ExpandAvalancheWrapper = styled.div`
    display: flex;
    gap: 6px;
    align-items: center;
`;
export const ExpandAvalancheLabel = styled.div`
    color: ${props => props.theme.eCashBlue};
    font-style: italic;
`;
export const ExpandButtonPanel = styled.div`
    display: flex;
`;
export const PanelButton = styled.button`
    border: none;
    background: none;
    cursor: pointer;
`;
export const PanelLink = styled(Link)`
    border: none;
    background: none;
    cursor: pointer;
`;
export const ReplyLink = styled(PanelLink)`
    margin-left: auto;
`;
export const AddressLink = styled.a`
    padding: 0 3px;
`;
export const AppAction = styled.div`
    display: flex;
    gap: 12px;
    width: 100%;
    align-items: center;
    justify-content: space-between;
    padding: 3px 12px;
    ${props => props.type === 'Received' && Incoming}
    border-radius: 9px;
    background-color: ${props => props.theme.panel};
    flex-wrap: wrap;
    word-break: break-all;
`;
export const AppDescLabel = styled.div`
    font-weight: bold;
`;
export const IconAndLabel = styled.div`
    display: flex;
    gap: 6px;
    align-items: center;
`;
export const AppDescMsg = styled.div`
    overflow-wrap: break-word;
    word-wrap: break-word;
    word-break: break-all;
    text-align: left;
`;
export const TokenAction = styled(AppAction)`
    ${props => props.tokenTxType === 'Received' && Incoming}
    ${props =>
        (props.tokenTxType === 'Created' || props.tokenTxType === 'Minted') &&
        Genesis}
        ${props => props.tokenTxType === 'Burned' && Burn}
`;
export const TokenActionHolder = styled.div``;
export const TokenInfoCol = styled.div`
    display: flex;
    flex-direction: column;
`;
export const UnknownMsgColumn = styled.div`
    display: flex;
    flex-direction: column;
    text-align: left;
`;
export const TokenType = styled.div``;
export const TokenName = styled.div``;
export const TokenTicker = styled.div``;
export const TokenDesc = styled.div``;
export const ActionLink = styled.a``;
