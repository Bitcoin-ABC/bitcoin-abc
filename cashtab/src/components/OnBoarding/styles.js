// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const WelcomeCtn = styled.div`
    margin-top: 20px;
    padding: 0px 30px;
    color: ${props => props.theme.contrast};
    h2 {
        color: ${props => props.theme.contrast};
    }
`;

export const WelcomeText = styled.p`
    width: 100%;
    font-size: 16px;
    margin-bottom: 60px;
    text-align: left;
`;

export const WelcomeLink = styled.a`
    text-decoration: underline;
    color: ${props => props.theme.eCashBlue};
    :hover {
        color: ${props => props.theme.eCashPurple} !important;
        text-decoration: underline !important;
    }
`;
