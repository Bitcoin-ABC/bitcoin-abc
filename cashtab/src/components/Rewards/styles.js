// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const Wrapper = styled.div`
    width: 100%;
    margin-top: 24px;
`;
