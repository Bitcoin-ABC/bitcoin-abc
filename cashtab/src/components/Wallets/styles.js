// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const WalletsList = styled.div`
    margin-top: 24px;
    padding: 12px;
    display: flex;
    flex-direction: column;
    width: 100%;
    align-items: center;
    gap: 12px;
    color: ${props => props.theme.contrast};
    box-sizing: border-box;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;

export const WalletsPanel = styled.div`
    display: flex;
    flex-direction: column;
    padding: 12px;
    width: 100%;
    background-color: ${props => props.theme.panel};
    border-radius: 9px;
    margin-bottom: 12px;
`;
export const Wallet = styled.div`
    display: flex;
    flex-direction: column;
    border-top: 0.5px solid ${props => props.theme.separator};
    gap: 0 12px;
    padding: 6px 0;
`;
export const WalletRow = styled.div`
    display: flex;
    width: 100%;
    align-items: center;
    justify-content: space-between;
`;
export const ActionsRow = styled.div``;

export const WalletName = styled.div`
    display: flex;
    text-align: left;
    word-break: break-word;
`;

export const ActiveWalletName = styled(WalletName)`
    font-weight: bold;
    color: ${props => props.theme.eCashBlue};
`;

export const SvgButtonPanel = styled.div`
    display: flex;
    align-items: baseline;
`;
export const ButtonPanel = styled.div`
    display: flex;
    gap: 9px;
    align-items: center;
    justify-content: center;
`;

export const WalletBalance = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    word-wrap: break-word;
    hyphens: auto;
`;

export const ActivateButton = styled.button`
    cursor: pointer;
    color: ${props => props.theme.eCashBlue};
    border-radius: 9px;
    border: 2px solid ${props => props.theme.eCashBlue};
    background: transparent;
    :hover {
        background-color: ${props => props.theme.eCashBlue};
        color: ${props => props.theme.contrast};
    }
`;
