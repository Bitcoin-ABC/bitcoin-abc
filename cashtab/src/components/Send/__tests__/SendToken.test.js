import React from 'react';
import renderer from 'react-test-renderer';
import { renderHook, act } from '@testing-library/react-hooks';
import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import SendToken from 'components/Send/SendToken';
import {
    walletWithBalancesAndTokens,
    walletWithBalancesAndTokensWithCorrectState,
} from '../../Home/__mocks__/walletAndBalancesMock';
import { WalletContext } from 'utils/context';
import { BrowserRouter as Router } from 'react-router-dom';

beforeEach(() => {
    // Mock method not implemented in JSDOM
    // See reference at https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation(query => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
});

test('Wallet with BCH balances and tokens', () => {
    const component = renderer.create(
        <WalletContext.Provider value={walletWithBalancesAndTokens}>
            <ThemeProvider theme={theme}>
                <Router>
                    <SendToken
                        tokenId={
                            'bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba'
                        }
                    />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Wallet with BCH balances and tokens and state field', () => {
    const component = renderer.create(
        <WalletContext.Provider
            value={walletWithBalancesAndTokensWithCorrectState}
        >
            <ThemeProvider theme={theme}>
                <Router>
                    <SendToken
                        tokenId={
                            'bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba'
                        }
                    />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Without wallet defined', () => {
    const withoutWalletDefinedMock = {
        wallet: {},
        balances: { totalBalance: 0 },
        loading: false,
    };
    const component = renderer.create(
        <WalletContext.Provider value={withoutWalletDefinedMock}>
            <ThemeProvider theme={theme}>
                <Router>
                    <SendToken
                        tokenId={
                            'bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba'
                        }
                    />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Verify the SendToken component successfully initializes', async () => {
    const { result } = renderHook(() => (
        // Wallet context mock needed here as the `SendToken(tokenId)` call won't instantiate without a wallet object present
        <WalletContext.Provider value={walletWithBalancesAndTokens}>
            SendToken(`bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba`)
        </WalletContext.Provider>
    ));

    // Extract component state at the point of initialization
    let initializedSendToken;
    await act(async () => {
        initializedSendToken = await result.current;
    });

    // Ensure component state is not null
    expect(initializedSendToken).not.toBeNull();

    // Ensure successful initialization of SendToken component props
    const expectedSendTokenProps = [
        'wallet',
        'balances',
        'tokens',
        'loading',
        'cashtabCache',
    ];
    const validSendTokenProps = expectedSendTokenProps.every(prop =>
        Object.prototype.hasOwnProperty.call(
            initializedSendToken.props.value,
            prop,
        ),
    );
    expect(validSendTokenProps).toBe(true);
});

test('Verify the SendToken component throws error on missing Wallet object', () => {
    const { result } = renderHook(() =>
        // No wallet context mock provided
        SendToken(
            `bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba`,
        ),
    );
    // React/Jest dynamically generates the `((cov_1erx5eord5(...).s[16]++)`
    // portion of the expected error message:
    // `Cannot destructure property 'wallet' of ((cov_1erx5eord5(...).s[16]++) _react.default.useContext(...)) as it is undefined.`
    // at the time of test execution, hence this error test case only checks that
    // the error message string contains `Cannot destructure property 'wallet'`
    expect(result.error.message).toEqual(
        expect.stringContaining(`Cannot destructure property 'wallet' of`),
    );
});

test('Verify the SendToken component throws error on missing token ID', async () => {
    const { result } = renderHook(() =>
        // No token id provided as input
        SendToken(),
    );
    expect(result.error.message).toBe(
        `Cannot destructure property 'tokenId' of 'undefined' as it is undefined.`,
    );
});
