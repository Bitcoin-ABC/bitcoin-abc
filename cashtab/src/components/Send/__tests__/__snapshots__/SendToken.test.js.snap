// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet with BCH balances and tokens 1`] = `null`;

exports[`Wallet with BCH balances and tokens and state field 1`] = `
<div
  className="sc-gPEVay dOhwxg"
>
  <div
    className="sc-csuQGl fFnofI"
  >
    6.001000000
     
    TBS
  </div>
  <div
    className="ant-row"
    style={Object {}}
    type="flex"
  >
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal"
        onReset={[Function]}
        onSubmit={[Function]}
        style={
          Object {
            "width": "auto",
          }
        }
      >
        <div
          className="sc-eHgmQL jlnegu"
        >
          <div
            className="ant-form-item ant-form-item-with-help"
            style={
              Object {
                "marginBottom": "0px",
              }
            }
          >
            <div
              className="ant-row ant-form-item-row"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <span
                      className="ant-input-group-wrapper"
                    >
                      <span
                        className="ant-input-wrapper ant-input-group"
                      >
                        <span
                          className="ant-input-affix-wrapper"
                          hidden={null}
                          onClick={[Function]}
                          style={null}
                        >
                          <span
                            className="ant-input-prefix"
                          >
                            <span
                              aria-label="wallet"
                              className="anticon anticon-wallet sc-bxivhb iBBRHU"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="wallet"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 464H528V448h312v128zm0 264H184V184h656v200H496c-17.7 0-32 14.3-32 32v192c0 17.7 14.3 32 32 32h344v200zM580 512a40 40 0 1080 0 40 40 0 10-80 0z"
                                />
                              </svg>
                            </span>
                          </span>
                          <input
                            autoComplete="off"
                            className="ant-input"
                            hidden={null}
                            name="address"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            placeholder="Address"
                            required={true}
                            style={null}
                            type="text"
                            value=""
                          />
                        </span>
                        <span
                          className="ant-input-group-addon"
                        >
                          <span
                            className="sc-kkGfuU cRGTNd"
                            onClick={[Function]}
                          >
                            <span
                              aria-label="qrcode"
                              className="anticon anticon-qrcode sc-ifAKCX jqgRJL"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="qrcode"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M468 128H160c-17.7 0-32 14.3-32 32v308c0 4.4 3.6 8 8 8h332c4.4 0 8-3.6 8-8V136c0-4.4-3.6-8-8-8zm-56 284H192V192h220v220zm-138-74h56c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm194 210H136c-4.4 0-8 3.6-8 8v308c0 17.7 14.3 32 32 32h308c4.4 0 8-3.6 8-8V556c0-4.4-3.6-8-8-8zm-56 284H192V612h220v220zm-138-74h56c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm590-630H556c-4.4 0-8 3.6-8 8v332c0 4.4 3.6 8 8 8h332c4.4 0 8-3.6 8-8V160c0-17.7-14.3-32-32-32zm-32 284H612V192h220v220zm-138-74h56c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm194 210h-48c-4.4 0-8 3.6-8 8v134h-78V556c0-4.4-3.6-8-8-8H556c-4.4 0-8 3.6-8 8v332c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V644h78v102c0 4.4 3.6 8 8 8h190c4.4 0 8-3.6 8-8V556c0-4.4-3.6-8-8-8zM746 832h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm142 0h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-eqIVtm khwohh"
        >
          <a
            className="sc-jDwBTQ dGROed"
            href="https://explorer.e.cash/address/false"
            rel="noreferrer"
            target="_blank"
          />
        </div>
        <br />
        <div
          className="sc-eHgmQL jlnegu"
        >
          <div
            className="ant-form-item ant-form-item-with-help"
          >
            <div
              className="ant-row ant-form-item-row"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <span
                      className="ant-input-group-wrapper"
                    >
                      <span
                        className="ant-input-wrapper ant-input-group"
                      >
                        <span
                          className="ant-input-affix-wrapper"
                          hidden={null}
                          onClick={[Function]}
                          style={null}
                        >
                          <span
                            className="ant-input-prefix"
                          />
                          <input
                            className="ant-input"
                            hidden={null}
                            name="value"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onWheel={[Function]}
                            placeholder="Amount"
                            required={true}
                            step={1e-9}
                            style={null}
                            type="number"
                            value=""
                          />
                          <span
                            className="ant-input-suffix"
                          >
                            TBS
                          </span>
                        </span>
                        <span
                          className="ant-input-group-addon"
                        >
                          <span
                            className="sc-cvbbAY losWJH"
                            disabled={false}
                            onClick={[Function]}
                          >
                            max
                          </span>
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingTop": "12px",
            }
          }
        >
          <button
            className="sc-jKJlTe jWeghZ"
            onClick={[Function]}
          >
            Send 
            TestBits
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Without wallet defined 1`] = `null`;
